rust/core-lib/src/movement.rs

```diff
@@ -49,6 +49,8 @@ pub enum Movement {
     StartOfParagraph,
     /// Move to the end of the text line.
     EndOfParagraph,
+    /// Move to the end of the text line, or next line if already at end.
+    EndOfParagraphKill,
```

Comment:
No, it's just that these two movement variations do two different things when the cursor is at the end of the line.

FWIW, we'll probably have a lot more movements to take care of vi.